--- image-httpd-ssl.conf	2020-08-26 15:01:49.000000000 -0700
+++ httpd-ssl.conf	2020-09-01 15:43:32.000000000 -0700
@@ -49,8 +49,10 @@
 #   ensure these follow appropriate best practices for this deployment.
 #   httpd 2.2.30, 2.4.13 and later force-disable aNULL, eNULL and EXP ciphers,
 #   while OpenSSL disabled these by default in 0.9.8zf/1.0.0r/1.0.1m/1.0.2a.
-SSLCipherSuite HIGH:MEDIUM:!MD5:!RC4:!3DES
-SSLProxyCipherSuite HIGH:MEDIUM:!MD5:!RC4:!3DES
+
+# Disables insecure encryption ciphers
+SSLCipherSuite ECDH+AESGCM:ECDH+CHACHA20:DH+AESGCM:ECDH+AES256:DH+AES256:ECDH+AES128:DH+AES:!RSA+AESGCM:!RSA+AES:!aNULL:!MD5:!DSS
+SSLProxyCipherSuite ECDH+AESGCM:ECDH+CHACHA20:DH+AESGCM:ECDH+AES256:DH+AES256:ECDH+AES128:DH+AES:!RSA+AESGCM:!RSA+AES:!aNULL:!MD5:!DSS
 
 #  By the end of 2016, only TLSv1.2 ciphers should remain in use.
 #  Older ciphers should be disallowed as soon as possible, while the
@@ -76,8 +78,10 @@
 #   Disable SSLv3 by default (cf. RFC 7525 3.1.1).  TLSv1 (1.0) should be
 #   disabled as quickly as practical.  By the end of 2016, only the TLSv1.2
 #   protocol or later should remain in use.
-SSLProtocol all -SSLv3
-SSLProxyProtocol all -SSLv3
+
+# Disables SSL and TLS older than 1.2:
+SSLProtocol all -SSLv2 -SSLv3 -TLSv1 -TLSv1.1
+SSLProxyProtocol all -SSLv2 -SSLv3 -TLSv1 -TLSv1.1
 
 #   Pass Phrase Dialog:
 #   Configure the pass phrase gathering process.
@@ -122,7 +126,7 @@
 
 #   General setup for the virtual host
 DocumentRoot "/usr/local/apache2/htdocs"
-ServerName www.example.com:443
+ServerName ${METACAT_UI_DOMAIN}:443
 ServerAdmin you@example.com
 ErrorLog /proc/self/fd/2
 TransferLog /proc/self/fd/1
@@ -141,7 +145,7 @@
 #   Some ECC cipher suites (http://www.ietf.org/rfc/rfc4492.txt)
 #   require an ECC certificate which can also be configured in
 #   parallel.
-SSLCertificateFile "/usr/local/apache2/conf/server.crt"
+SSLCertificateFile ${SSL_SERVER_CERT}
 #SSLCertificateFile "/usr/local/apache2/conf/server-dsa.crt"
 #SSLCertificateFile "/usr/local/apache2/conf/server-ecc.crt"
 
@@ -151,7 +155,7 @@
 #   you've both a RSA and a DSA private key you can configure
 #   both in parallel (to also allow the use of DSA ciphers, etc.)
 #   ECC keys, when in use, can also be configured in parallel
-SSLCertificateKeyFile "/usr/local/apache2/conf/server.key"
+SSLCertificateKeyFile ${SSL_SERVER_KEY}
 #SSLCertificateKeyFile "/usr/local/apache2/conf/server-dsa.key"
 #SSLCertificateKeyFile "/usr/local/apache2/conf/server-ecc.key"
 
@@ -162,7 +166,8 @@
 #   the referenced file can be the same as SSLCertificateFile
 #   when the CA certificates are directly appended to the server
 #   certificate for convenience.
-#SSLCertificateChainFile "/usr/local/apache2/conf/server-ca.crt"
+SSLCertificateChainFile ${SSL_SERVER_CHAIN}
+
 
 #   Certificate Authority (CA):
 #   Set the CA certificate verification path where to find CA
@@ -172,7 +177,7 @@
 #         to point to the certificate files. Use the provided
 #         Makefile to update the hash symlinks after changes.
 #SSLCACertificatePath "/usr/local/apache2/conf/ssl.crt"
-#SSLCACertificateFile "/usr/local/apache2/conf/ssl.crt/ca-bundle.crt"
+SSLCACertificateFile "/usr/local/apache2/conf/ssl.crt/ca-bundle.crt"
 
 #   Certificate Revocation Lists (CRL):
 #   Set the CA revocation path where to find CA CRLs for client
@@ -284,7 +289,42 @@
 #   Per-Server Logging:
 #   The home of a custom SSL log file. Use this when you want a
 #   compact non-error SSL logfile on a virtual host basis.
-CustomLog /proc/self/fd/1 \
+#CustomLog /proc/self/fd/1 \
           "%t %h %{SSL_PROTOCOL}x %{SSL_CIPHER}x \"%r\" %b"
 
-</VirtualHost>                                  
+# Forward the Client Certificate to Tomcat
+SSLVerifyClient optional
+SSLVerifyDepth 10
+SSLOptions +StdEnvVars +StrictRequire +ExportCertData
+
+# Turns on HSTS (we should be careful about production rollout here; 
+# it forces an internal, browser-based redirect to HTTPS after 
+# it is encountered (!):
+Header always set Strict-Transport-Security "max-age=31536000;"
+
+
+AllowEncodedSlashes On
+AcceptPathInfo      On
+
+# AddDefaultCharset will be needed when exporting
+# cert data to the proxy
+<Proxy *>
+     AddDefaultCharset Off
+</Proxy>
+
+ProxyPass "/${METACAT_APP_CONTEXT}/" "ajp://${METACAT_MN_DOMAIN}:8009/${METACAT_APP_CONTEXT}/" nocanon retry=0 timeout=300
+ProxyPassReverse "/${METACAT_APP_CONTEXT}/" "ajp:/${METACAT_MN_DOMAIN}:8009/${METACAT_APP_CONTEXT}/"
+
+<IfDefine EnableReverseProxy>
+
+    RemoteIPProxyProtocol On
+    
+    # Requests to this virtual host must have a PROXY protocol
+    # header provided. If it is missing, the connection will
+    # be aborted
+
+</IfDefine>
+
+
+
+</VirtualHost>
