--- image-httpd.conf	2019-09-16 13:19:48.000000000 -0700
+++ httpd.conf	2019-08-19 21:31:50.000000000 -0700
@@ -89,12 +89,11 @@
 #LoadModule isapi_module modules/mod_isapi.so
 #LoadModule file_cache_module modules/mod_file_cache.so
 #LoadModule cache_module modules/mod_cache.so
-#LoadModule cache_disk_module modules/mod_cache_disk.so
+#LoadModule cache_disk_module modules/mod4_cache_disk.so
 #LoadModule cache_socache_module modules/mod_cache_socache.so
-#LoadModule socache_shmcb_module modules/mod_socache_shmcb.so
+LoadModule socache_shmcb_module modules/mod_socache_shmcb.so
 #LoadModule socache_dbm_module modules/mod_socache_dbm.so
 #LoadModule socache_memcache_module modules/mod_socache_memcache.so
-#LoadModule socache_redis_module modules/mod_socache_redis.so
 #LoadModule watchdog_module modules/mod_watchdog.so
 #LoadModule macro_module modules/mod_macro.so
 #LoadModule dbd_module modules/mod_dbd.so
@@ -120,7 +119,6 @@
 #LoadModule deflate_module modules/mod_deflate.so
 #LoadModule xml2enc_module modules/mod_xml2enc.so
 #LoadModule proxy_html_module modules/mod_proxy_html.so
-#LoadModule brotli_module modules/mod_brotli.so
 LoadModule mime_module modules/mod_mime.so
 #LoadModule ldap_module modules/mod_ldap.so
 LoadModule log_config_module modules/mod_log_config.so
@@ -138,17 +136,17 @@
 #LoadModule unique_id_module modules/mod_unique_id.so
 LoadModule setenvif_module modules/mod_setenvif.so
 LoadModule version_module modules/mod_version.so
-#LoadModule remoteip_module modules/mod_remoteip.so
-#LoadModule proxy_module modules/mod_proxy.so
+LoadModule remoteip_module modules/mod_remoteip.so
+LoadModule proxy_module modules/mod_proxy.so
 #LoadModule proxy_connect_module modules/mod_proxy_connect.so
 #LoadModule proxy_ftp_module modules/mod_proxy_ftp.so
-#LoadModule proxy_http_module modules/mod_proxy_http.so
+LoadModule proxy_http_module modules/mod_proxy_http.so
 #LoadModule proxy_fcgi_module modules/mod_proxy_fcgi.so
 #LoadModule proxy_scgi_module modules/mod_proxy_scgi.so
 #LoadModule proxy_uwsgi_module modules/mod_proxy_uwsgi.so
 #LoadModule proxy_fdpass_module modules/mod_proxy_fdpass.so
 #LoadModule proxy_wstunnel_module modules/mod_proxy_wstunnel.so
-#LoadModule proxy_ajp_module modules/mod_proxy_ajp.so
+LoadModule proxy_ajp_module modules/mod_proxy_ajp.so
 #LoadModule proxy_balancer_module modules/mod_proxy_balancer.so
 #LoadModule proxy_express_module modules/mod_proxy_express.so
 #LoadModule proxy_hcheck_module modules/mod_proxy_hcheck.so
@@ -158,7 +156,7 @@
 #LoadModule session_dbd_module modules/mod_session_dbd.so
 #LoadModule slotmem_shm_module modules/mod_slotmem_shm.so
 #LoadModule slotmem_plain_module modules/mod_slotmem_plain.so
-#LoadModule ssl_module modules/mod_ssl.so
+LoadModule ssl_module modules/mod_ssl.so
 #LoadModule optional_hook_export_module modules/mod_optional_hook_export.so
 #LoadModule optional_hook_import_module modules/mod_optional_hook_import.so
 #LoadModule optional_fn_import_module modules/mod_optional_fn_import.so
@@ -166,7 +164,6 @@
 #LoadModule dialup_module modules/mod_dialup.so
 #LoadModule http2_module modules/mod_http2.so
 #LoadModule proxy_http2_module modules/mod_proxy_http2.so
-#LoadModule md_module modules/mod_md.so
 #LoadModule lbmethod_byrequests_module modules/mod_lbmethod_byrequests.so
 #LoadModule lbmethod_bytraffic_module modules/mod_lbmethod_bytraffic.so
 #LoadModule lbmethod_bybusyness_module modules/mod_lbmethod_bybusyness.so
@@ -238,7 +235,7 @@
 #
 # If your host doesn't have a registered DNS name, enter its IP address here.
 #
-#ServerName www.example.com:80
+ServerName ${METACAT_UI_DOMAIN}:80
 
 #
 # Deny access to the entirety of your server's filesystem. You must
@@ -264,6 +261,14 @@
 #
 DocumentRoot "/usr/local/apache2/htdocs"
 <Directory "/usr/local/apache2/htdocs">
+
+    # The MetacatUI new routing changes work because when a client navigates 
+    # to a MetacatUI view via a full pathname (/data) rather than a hash 
+    # (/#data), Apache will serve the MetacatUI index.html page as a fallback 
+    # resource instead of returning a 404 error. The fallback resource is 
+    # specified via this Apache setting:
+    FallbackResource /index.html
+ 
     #
     # Possible values for the Options directive are "None", "All",
     # or any combination of:
@@ -276,7 +281,7 @@
     # http://httpd.apache.org/docs/2.4/mod/core.html#options
     # for more information.
     #
-    Options Indexes FollowSymLinks
+    Options FollowSymLinks
 
     #
     # AllowOverride controls what directives may be placed in .htaccess files.
@@ -324,6 +329,9 @@
 LogLevel warn
 
 <IfModule log_config_module>
+
+    LogLevel info ssl:warn
+
     #
     # The following directives define some format nicknames for use with
     # a CustomLog directive (see below).
@@ -343,13 +351,18 @@
     # define per-<VirtualHost> access logfiles, transactions will be
     # logged therein and *not* in this file.
     #
-    CustomLog /proc/self/fd/1 common
+    #CustomLog /proc/self/fd/1 common
 
     #
     # If you prefer a logfile with access, agent, and referer information
     # (Combined Logfile Format) you can use the following directive.
     #
     #CustomLog "logs/access_log" combined
+    
+    # The log format is determined by the most recently specified 
+    # LogFormat directive which does not define a nickname.
+    LogFormat "%a %l %u %t \"%r\" %>s %b \"%{Referer}i\" \"%{User-Agent}i\"" 
+    TransferLog /proc/self/fd/1
 </IfModule>
 
 <IfModule alias_module>
@@ -537,8 +550,12 @@
 Include conf/extra/proxy-html.conf
 </IfModule>
 
+
+<IfDefine EnableSSL>
 # Secure (SSL/TLS) connections
-#Include conf/extra/httpd-ssl.conf
+Include conf/extra/httpd-ssl.conf
+</IfDefine>
+
 #
 # Note: The following must must be present to support
 #       starting without SSL on platforms with no /dev/random equivalent
@@ -549,3 +566,24 @@
 SSLRandomSeed connect builtin
 </IfModule>
 
+Options -Indexes
+
+<VirtualHost *:80>
+
+    AllowEncodedSlashes On
+    AcceptPathInfo      On
+
+    ProxyPass "/${METACAT_APP_CONTEXT}/" "ajp://${METACAT_MN_DOMAIN}:8009/${METACAT_APP_CONTEXT}/" nocanon retry=0 timeout=300
+    ProxyPassReverse "/${METACAT_APP_CONTEXT}/" "ajp:/${METACAT_MN_DOMAIN}:8009/${METACAT_APP_CONTEXT}/"
+
+    <IfDefine EnableReverseProxy>
+
+        RemoteIPProxyProtocol On
+        
+        # Requests to this virtual host must have a PROXY protocol
+        # header provided. If it is missing, the connection will
+        # be aborted
+
+    </IfDefine>
+
+</VirtualHost>
